apps/api/src/users/users.controller.ts

```diff
@@ -36,4 +52,74 @@ export class UsersController {
   ): Promise<AccessTokenResponse> {
     return this.usersService.updateUserPassword(user, updateUserPasswordDto);
   }
+
+  @UseGuards(JwtAuthGuard)
+  @Header('Content-Type', 'image/jpeg')
+  @Get(':id/profile-picture')
+  async getUserProfilePicture(
+    @Param('id') id: string,
+  ): Promise<StreamableFile> {
+    const profile = await this.usersService.getProfile(id);
+    if (profile?.picture == null) {
+      throw new NotFoundException(
+        `User: ${id} does not have a profile picture`,
+      );
+    }
+    const file = fs.createReadStream(profile.picture);
+    return new StreamableFile(file);
+  }
+
+  @UseGuards(JwtAuthGuard)
+  @Put(':id/update-profile-picture')
+  @UseInterceptors(
+    FileInterceptor('file', {
+      storage: diskStorage({
+        destination: './uploads/profile_pictures/',
+        filename: (req, file, cb) => {
+          const filename: string =
+            path.parse(file.originalname).name.replace(/\s/g, '') +
+            randomUUID();
+          const extension: string = path.parse(file.originalname).ext;
+          cb(null, `${filename}${extension}`);
+        },
+      }),
+    }),
+  )
+  @HttpCode(204)
+  async updateProfilePicture(
+    @Param('id') id: string,
```

Comment:
I think that the _id_ of the user should be retrieved from its JWT. Here, whoever is logged in can change the profile picture of anyone.